sturdycouscous/Printer.py

```diff
@@ -0,0 +1,173 @@
+from Mongo_Printer_Client import DBClient
+from models import TLS_Record
+
+class Printer():
+    def __init__(self):
+        self.client = DBClient()
+        self.outputfile = open('output.txt', "w")
+
+    def write_to_file(self, output):
+        pass
+        self.outputfile.writelines(output)
+        self.outputfile.close()
+
+    def mongo_query(self, fields):
+        output = []
+        for row in self.client.column.find({}, fields):
+            output.append(row)
+        print(output)
+        return output
+
+    def dump_db(self):
+        for row in self.client.column.find({}, {'url', 'title'}):
+            record = tls_record().parse_record(row)
+            write_to_file(record.dump_dict())
+
+    def print_record_count(self):
+        return len(self.mongo_query({'_id'}))
+
+    def tally_open_ports(self):
+        tally = {}
+        query_filter = {'Checker.open_ports'}
+
+        # Retrieve data and tally results
+        for row in self.mongo_query(query_filter):
+            ports = row['Checker']['open_ports']
+            for port in ports:
+                if port not in tally:
+                    tally[port] = 1
+                else:
+                    tally[port] += 1
```

Comment:
Let's use the power of Mongo!

```suggestion
        for tsl_version in  ["TLSv1.3", "TLSv1.2", "TLSv1.1", "TLSv1.0"]:
            for port in  [20, 21, 69, 80, 123, 8080, 389]
                tally[port] = self.collection.count_documents({ 'Checker.port': tls_version})
```